### PR TITLE
Adds mx.nd.moveaxis

### PR DESCRIPTION
**Pull request by JeanKossaifi**
Adds a moveaxis function for ndarrays.

### CODE REVIEW

**Comment by piiswrong** [comment]
we have swapaxes for this

**Comment by JeanKossaifi** [comment]
`swapaxis` and `moveaxis` are different: the former one swaps the position of two dimensions while the latter moves one axis from one position to another. 
I am using the same naming convention as NumPy which offers both `swapaxes` and `moveaxes`.

File: python/mxnet/ndarray.py [hunk 1]
```diff
@@ -1054,6 +1054,45 @@ def array(source_array, ctx=None, dtype=None):
...
+    (3, 2)
+    """
+    axes = list(range(tensor.ndim))
+    try:
+        axes.pop(source)
+    except IndexError:
+        raise ValueError('Source should verify 0 <= source < tensor.ndim'
+                         'Got %d' % source)
+    try:
+        axes.insert(destination, source)
```
Comment (piiswrong) [inline python/mxnet/ndarray.py]: This will fail with moveaxis(x, [0, 1, 2], [2, 1, 0]) where x is 3d array

Comment (JeanKossaifi) [inline python/mxnet/ndarray.py]: I wrote the function to take `ints` as parameters and not lists so as to not overload the function. I can add support for lists of indices as well if you prefer.

Comment (piiswrong) [inline python/mxnet/ndarray.py]: Ok. Then please modify the doc

Comment (JeanKossaifi) [inline python/mxnet/ndarray.py]: Done.

**Pull request merged**